client/layout/masterbar/oauth-client.jsx

```diff
@@ -1,32 +1,47 @@
-import { Gridicon } from '@automattic/components';
+import { AkismetLogo, GravatarLogo, Gridicon } from '@automattic/components';
 import { localizeUrl } from '@automattic/i18n-utils';
 import PropTypes from 'prop-types';
 import JetpackLogo from 'calypso/components/jetpack-logo';
 import {
+	isAkismetOAuth2Client,
 	isCrowdsignalOAuth2Client,
+	isGravatarOAuth2Client,
 	isWooOAuth2Client,
 	isJetpackCloudOAuth2Client,
 } from 'calypso/lib/oauth2-clients';
 import CrowdsignalOauthMasterbar from './crowdsignal';
 import WooOauthMasterbar from './woo';
 import './oauth-client.scss';
 
+const OauthClientLogo = ( { oauth2Client } ) => {
+	if ( isJetpackCloudOAuth2Client( oauth2Client ) ) {
+		return <JetpackLogo full monochrome={ false } size={ 28 } />;
+	}
+
+	if ( isAkismetOAuth2Client( oauth2Client ) ) {
+		return <AkismetLogo size={ 28 } />;
+	}
+
+	if ( isGravatarOAuth2Client( oauth2Client ) ) {
+		return <GravatarLogo size={ 28 } />;
+	}
+
+	// Other clients can define their own logos
+	if ( oauth2Client.icon ) {
+		return <img src={ oauth2Client.icon } alt={ oauth2Client.title || '' } />;
```

Comment:
I wonder if there's a need to fall back to an empty string. Won't every client eventually have a title set?

```suggestion
		return <img src={ oauth2Client.icon } alt={ oauth2Client.title } />;
```